src/OpenCensus.Collector.Dependencies/DependenciesCollector.cs

```diff
@@ -40,10 +41,14 @@ public class DependenciesCollector : IDisposable
         public DependenciesCollector(DependenciesCollectorOptions options, ITracer tracer, ISampler sampler, IPropagationComponent propagationComponent)
         {
             this.diagnosticSourceSubscriber = new DiagnosticSourceSubscriber(
-                new Dictionary<string, Func<ITracer, ISampler, ListenerHandler>>()
+                new Dictionary<string, Func<ITracer, Func<HttpRequestMessage, ISampler>, ListenerHandler>>()
                 { { "HttpHandlerDiagnosticListener", (t, s) => new HttpHandlerDiagnosticListener(t, s, propagationComponent) } },
                 tracer,
-                sampler);
+                x =>
+                {
+                    ISampler s = options.CustomSampler(x);
```

Comment:
catch exception here. If method failed with exception - default sampler should be used

Reply:
Done. Q - If I put in a Debug.WriteLine(..) here in exception case, where would it emerge, would the caller be able to see it in event viewer ?

Reply:
It may be outside of a scope, but we will need to log error in Event Source https://github.com/census-instrumentation/opencensus-csharp/blob/develop/docs/error-logging.md in future,

Reply:
Marked as TODO